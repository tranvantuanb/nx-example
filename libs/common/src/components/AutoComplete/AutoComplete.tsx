import React, { useMemo } from 'react';
import {
  AutoComplete as AntAutoComplete,
  AutoCompleteProps as AntAutoCompleteProps,
} from 'antd';
import tw, { styled } from 'twin.macro';

interface OptionType {
  label?: React.ReactNode;
  title?: React.ReactNode;
  id?: string | number | null | undefined;
  value?: string | number | null | undefined;
}

export type AutoCompleteProps = AntAutoCompleteProps<any, OptionType>;

interface IAutoComplete extends React.FC<AutoCompleteProps> {
  Option: typeof AntAutoComplete.Option;
}

const StyledAutoComplete = styled(
  AntAutoComplete
)`` as unknown as typeof AntAutoComplete;

const AutoComplete: IAutoComplete = ({ options, ...props }) => {
  const transformedOptions = useMemo(
    () =>
      options?.map((item) => ({
        label: item.title,
        value: item.id,
        ...item,
      })),
    [options]
  );
  return <StyledAutoComplete options={transformedOptions} {...props} />;
};

AutoComplete.Option = AntAutoComplete.Option;

export { AutoComplete };
