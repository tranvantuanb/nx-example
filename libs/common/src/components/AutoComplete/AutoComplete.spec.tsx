import { render } from '../../test';
import { AutoComplete, AutoCompleteProps } from './AutoComplete';

describe('AutoComplete', () => {
  it('should render successfully', () => {
    const defaultProps: AutoCompleteProps = {};
    const { container } = render(<AutoComplete {...defaultProps} />);
    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
