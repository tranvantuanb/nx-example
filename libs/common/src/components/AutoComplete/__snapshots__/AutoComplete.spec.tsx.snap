// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoComplete should render successfully 1`] = `
<div>
  <div
    class="ant-select ant-select-auto-complete css-9bhhxd-StyledAutoComplete epi6ux40 ant-select-single ant-select-show-search"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          role="combobox"
          type="search"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      />
    </div>
  </div>
</div>
`;
